File: vortexdbg-api/src/main/java/com/vortexdbg/debugger/Breaker.java
package com.vortexdbg.debugger;

import com.vortexdbg.pointer.VortexdbgPointer;

public interface Breaker {

    default void debug() {
        debug(null);
    }

    void debug(String reason);

    void brk(VortexdbgPointer pc, int svcNumber);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/debugger/Breaker.kt
package com.vortexdbg.debugger

import com.vortexdbg.pointer.VortexdbgPointer

interface Breaker {

    fun debug() {
        debug("")
    }

    fun debug(reason: String)

    fun brk(pc: VortexdbgPointer?, svcNumber: Int)

}
